app/models/nic/bond.rb

```diff
@@ -44,6 +45,17 @@ def ensure_virtual
     def enc_attributes
       @enc_attributes ||= (super + %w(mode attached_devices bond_options))
     end
+
+    def ensure_mac
+      mac_addresses = attached_devices_identifiers.collect { |a|
+        self.host.interfaces.where(identifier: a).first.mac
+      }
+      self.mac = nil unless mac_addresses.include? self.mac
+      unless self.mac
+        self.mac = mac_addresses.first
+      end
+      self.update_column(:mac, self.mac)
```

Comment:
if we move it to before validation, it will be saved in normal save chain (within transaction)